setup.py

```diff
@@ -934,6 +934,8 @@ def init_inc_lib_dirs(self):
             # directories with whitespace in the name to store libraries.
             cflags, ldflags = sysconfig.get_config_vars(
                     'CFLAGS', 'LDFLAGS')
+            cflags = cflags + ' ' + ('',os.environ.get('CC'))[os.environ.get('CC') != None]
+            ldflags = ldflags + ' ' + ('',os.environ.get('LDSHARED'))[os.environ.get('LDSHARED') != None]
```

Comment:
do you know what this part is for?

Reply:
I guess this appends the contents of `$CC` and `$LDSHARED` to `cflags` and `ldflags`, if those environment variables are set.

I guess the tricky construct here is that `os.environ.get()` returns `None` if the variable isn't set, and you can't concatenate `None` with a string. Instead it creates a tuple with an empty string and `os.environ.get()`, and picks either tuple member 0 or 1 depending on the `os.environ.get() != None` comparison.

For my usecase, these two lines aren't necessary, but I brought them along from the original patch.

Reply:
I was confused why CC and LDSHARED would be of use here

> For my usecase, these two lines aren't necessary, but I brought them along from the original patch.

ok, could you drop them?

Reply:
The point here is that the code iterates through `$CFLAGS` and `$LDFLAGS` and looks for any `-I` and `-L` options. I guess it's plausible to have a case where you'd pass `CC="compiler -Isomedir"` too, and this lets you pick that up - so I guess there's some reason why @Alexpux added it there in the first case?

Reply:
I see, thanks. I'd still prefer to drop it as I think we can expect users to pass these things via CFLAGS/LDFLAGS if needed.

Reply:
Ok, fair enough - let's drop it for now.